### PR TITLE
✨ 🌡️ add prometheus metrics exporter basics

### PR DESCRIPTION
**Pull request by goern**
Signed-off-by: Christoph Görn <goern@b4mad.net>

### CODE REVIEW

File: components/paddock/paddock/settings.py [hunk 1]
```diff
@@ -115,26 +121,35 @@
...
-        "readonly": env.db_url("READONLY_DATABASE_URL"),
+        "default": {
+            "ENGINE": "django_prometheus.db.backends.postgresql",
+            "NAME": os.getenv("DB_NAME", "paddock"),
+            "USER": os.getenv("DB_USER", ""),
+            "PASSWORD": os.getenv("DB_PASSWORD", ""),
+            "HOST": os.getenv("DB_HOST", ""),
+            "PORT": os.getenv("DB_PORT", "5432"),
+        },
+        "readonly": {
```
Comment (goern) [inline components/paddock/paddock/settings.py:145]: @durandom I am unsure how this gets created, not via https://github.com/b4mad/racing/blob/main/manifests/base/postgresql/cluster.yaml#L82

File: components/paddock/telemetry/models.py [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
 from picklefield.fields import PickledObjectField
+
+from django.db import models
+from django_prometheus.models import ExportModelOperationsMixin
+
 import datetime
 
 
-class Driver(models.Model):
+class Driver(ExportModelOperationsMixin("driver"), models.Model):
```
Comment (goern) [inline components/paddock/telemetry/models.py:10]: @durandom I did not add metrics of all models, let me know wyt

File: manifests/base/paddock/paddock-settings.yaml [hunk 3]
```diff
@@ -1,19 +1,18 @@
...
-    DJANGO_SUPERUSER_EMAIL: info@b4mad.racing
-    DJANGO_SUPERUSER_USERNAME: admin
-    B4MAD_RACING_MQTT_USER: crewchief
-    B4MAD_RACING_MQTT_PASSWORD: crewchief
-    B4MAD_RACING_MQTT_HOST: mosquitto-tcp
-    B4MAD_RACING_MQTT_PORT: "1883"
-    B4MAD_RACING_INFLUX_ORG: b4mad
-    B4MAD_RACING_INFLUX_TOKEN: 41e593a879202f9ebc0e5bc6f1b18a40e0a0d855
-    B4MAD_RACING_INFLUX_URL: http://influxdb2/
-    READONLY_DATABASE_URL: postgresql://reader:995f43ad3dc7ce93c4ae7504b4d7c229deb7c5ae@db-primary:5432/paddock
```
Comment (goern) [inline manifests/base/paddock/paddock-settings.yaml:17]: This file is not even applied, how got the ENV created?

File: manifests/base/paddock/deployment_config.yaml [hunk 4]
```diff
@@ -58,11 +58,31 @@ spec:
           terminationMessagePath: /dev/termination-log
           terminationMessagePolicy: File
         - env:
-            - name: DATABASE_URL
+            - name: DB_HOST
               valueFrom:
                 secretKeyRef:
                   name: db-pguser-paddock
-                  key: uri
+                  key: host
```
[the remark this answers is not in the document]
Comment (durandom) [reply manifests/base/paddock/deployment_config.yaml:65]: gotcha. If this is the reason.

File: manifests/base/paddock/paddock-settings.yaml (hunk 3 above)
Comment (durandom) [reply manifests/base/paddock/paddock-settings.yaml:17]: it's created in `env/phobos` - this is just here for reference

**Review by durandom — APPROVED** [review]

**Pull request merged**